mibidata/color.py

```diff
@@ -279,28 +254,28 @@ def compose_overlay(image, overlay_settings):
             break
         raise ValueError("Unexpected format of overlay_settings dictionary.")
 
-    for i, channel in enumerate(overlay_settings):
+    overlay = None
+    for channel in overlay_settings:
         item = overlay_settings[channel]
-        int_array = image[channel]
+        array = image[channel]
         # Because we treat the min differently, don't use np.clip
         range_min = item.get('intensity_lower', 0)
-        range_max = item.get('intensity_higher', int_array.max())
-        int_array[int_array > range_max] = range_max
-        int_array[int_array < range_min] = 0
-        float_array = int_array / float(range_max)
-        float_array[int_array > 0] += item.get('brightness', 0)
+        range_max = item.get('intensity_higher', array.max())
+        array[array > range_max] = range_max
+        array[array < range_min] = 0
+        array = array / float(range_max)
+        array[array > 0] += item.get('brightness', 0)
         # Apply default filter even if no blurring
-        ndimage.filters.convolve(float_array, kernel, output=float_array)
+        array = ndimage.filters.convolve(array, kernel)
```

Comment:
Very minor, but now there is a mix of return and inplace array operations here and in the next few lines.

Reply:
That was intentional, using the same array as input and output in `ndimage.filters.convolve` changes the data as it's processing and results in a different output than desired.